dotnet/samples/KernelSyntaxExamples/RepoUtils/EnumerableExtensions.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+
+namespace RepoUtils;
+
+public static class EnumerableExtensions
+{
+    public static IEnumerable<List<TSource>> ChunkByAggregate<TSource, TAccumulate>(
```

Comment:
Is there a chance that this method will be useful for developers as well? If yes, maybe we can ship it as part of `TextChunker`, with name something like `TextChunker.SplitParagraphsByAggregate` (not as extension, just static method). We can add some tests, mark it as experimental for the first period.

Otherwise, if this method is for example purposes only, maybe we can add it as private method in Example76_TextEmbedding to avoid creating separate files. I'm not sure this method will be re-used in other examples.

Reply:
It would probably be helpful, though the usage here is less related to text chunking and more so transforming for a unique HTTP request. Not sure it belongs in the TextChunker or in the scope of this PR. Thinking an example is sufficient for now?

Having a dedicated extension method allows for a nicer code flow via method chaining imo. I made it a separate file due to not being able to nest extension classes and the lack of the `file private` modifier in .NET 6 (C# 10). If it was a separate class in the file all examples could still call the extension and potentially cause some confusion over where it lives.

Thoughts?